doc/architectural/central-data-structures.md

```diff
@@ -0,0 +1,91 @@
+# Central Data Structures
+
+The following is some light technical documentation of the major data structures
+represented in the input transformation to Intermediate Representation (IR) inside
+CBMC and the assorted CProver tools.
+
+## goto_modelt
+
+The `goto_modelt` is the main data structure that CBMC (and the other tools) use to
+represent GOTO-IR (the `GOTO` Intermediate Representation).
+
+A `goto_modelt` is effectively a product type, consisting of:
```

Comment:
Do we expect readers of our documentation to know what a product type is?

Reply:
I was operating under the assumption that this is developer's documentation, and I expect CBMC developers to be exposed to Programming Language Theory, and thus, know what a product type is.

Is this an unsafe assumption to make?

If yes, I need to rework this and some other documentation that I'm preparing.

Reply:
"pair" is a much more easily understandable word to use explain this than "product type" in my opinion.

Reply:
"product" is no longer used.